pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1063,6 +1063,7 @@ func (c *Controller) syncNodes(
 			go func(idx int) {
 				defer createWait.Done()
 				nodeName := nodesNeedingStoragePods[idx]
+				podTemplates[idx].Labels[constants.OperatorLabelManagedByKey] = constants.OperatorLabelManagedByValue
```

Comment:
Instead of adding the labels here, let's add to the template itself - https://github.com/libopenstorage/operator/blob/master/pkg/controller/storagecluster/storagecluster.go#L1367.

Reply:
I actually tried adding it there and also here:
https://github.com/libopenstorage/operator/blob/master/pkg/controller/storagecluster/storagecluster.go#L1247-L1251

However, the storage node creation tests kept failing as a result. The only way I could maintain functionality was adding the label right before pod creation.

I think it has to do with the selector labels and how they're added.